nd4j/nd4j-tools/src/main/java/org/nd4j/cli/TFtoFlatFileConverter.java

```diff
@@ -0,0 +1,148 @@
+                                                                        .withInputSubgraph(0, SubGraphPredicate
+                                                                                        .withRoot(OpPredicate
+                                                                                                        .nameMatches(".*/dropout/random_uniform/mul"))
+                                                                                        .withInputSubgraph(0,
+                                                                                                        SubGraphPredicate
+                                                                                                                        .withRoot(OpPredicate
+                                                                                                                                        .nameMatches(".*/dropout/random_uniform/RandomUniform")))
+                                                                                        .withInputSubgraph(1,
+                                                                                                        SubGraphPredicate
+                                                                                                                        .withRoot(OpPredicate
+                                                                                                                                        .nameMatches(".*/dropout/random_uniform/sub")))
+
+                                                                        ))));
+
+        List<SubGraph> subGraphs = GraphTransformUtil.getSubgraphsMatching(sd, p);
+        int subGraphCount = subGraphs.size();
+        sd = GraphTransformUtil.replaceSubgraphsMatching(sd, p, new SubGraphProcessor() {
+            @Override
+            public List<SDVariable> processSubgraph(SameDiff sd, SubGraph subGraph) {
+                List<SDVariable> inputs = subGraph.inputs(); // Get inputs to the subgraph
+                // Find pre-dropout input variable:
+                SDVariable newOut = null;
+                for (SDVariable v : inputs) {
+                    if (v.getVarName().endsWith("/BiasAdd") || v.getVarName().endsWith("/Softmax")
+                                    || v.getVarName().endsWith("/add_1") || v.getVarName().endsWith("/Tanh")) {
+                        newOut = v;
+                        break;
+                    }
+                }
+
+                if (newOut != null) {
+                    // Pass this input variable as the new output
+                    return Collections.singletonList(newOut);
+                }
+
+                throw new RuntimeException("No pre-dropout input variable found");
+            }
+        });
+
+
+        System.out.println("Exporting file " + outFile);
+        sd.asFlatFile(new File(outFile));
+    }
+
+
+    @Test
+    @Ignore
+    public void convertFile() {
```

Comment:
If we're going to have a test, let's put it in nd4j-tests - and not have it ignored.

Reply:
Done